test/test_torch.py

```diff
@@ -8775,8 +8326,8 @@ def validate_setting(x):
 
         # test invalid index fails
         reference = torch.empty(10, device=device)
-        # can't test cuda because it is a device assert
-        if not reference.is_cuda:
+        # only tests asserts on cpu
+        if 'cpu' in device:
```

Comment:
`torch.tensor([0], device=device).device == 'cpu'` (you can create a wrapper function for this if you want.

Reply:
Sure you don't want to just pass actual devices?

Yeah I'll cut this PR up. Git doesn't do a great job showing these diffs.